common/lib/protocol-definitions/src/clients.ts

```diff
@@ -5,53 +5,137 @@
+ * connection mode is set to "write".
+ */
 export type ConnectionMode = "write" | "read";
 
+/**
+ * TODO
+ */
 export interface ICapabilities {
+	/**
+	 * Indicates if the client represents a potentially interactive session with a user (if 'true') or if it's a "system entity" (if 'false').
+	 *
+	 * @remarks
+	 *
+	 * The only "system entity" scenario at the moment is the
+	 * {@link https://fluidframework.com/docs/concepts/summarizer/ | summarizer client}.
+	 */
 	interactive: boolean;
 }
+
+/**
+ * TODO
+ */
 export interface IClientDetails {
+	/**
+	 * {@inheritDoc ICapabilities}
+	 */
 	capabilities: ICapabilities;
+
+	/**
+	 * TODO: What is this? Are there specific expected values? What does it mean for this to be undefined?
+	 */
 	type?: string;
+
 	/**
 	 * If the environment needs to specify multiple properties which gives info about the environment, then
-	 * it should be in particular format like: "prop1:val1;prop2:val2;prop3:val3"
+	 * it should be in particular format like: `prop1:val1;prop2:val2;prop3:val3`.
+	 *
+	 * TODO: What does it mean for this to be undefined?
 	 */
 	environment?: string;
+
+	/**
+	 * TODO: What is this? What does it mean for this to be undefined?
+	 */
 	device?: string;
 }
 
+/**
+ * TODO
+ */
 export interface IClient {
+	/**
+	 * {@inheritDoc ConnectionMode}
+	 */
 	mode: ConnectionMode;
+
+	/**
+	 * {@inheritDoc IClientDetails}
+	 */
 	details: IClientDetails;
+
+	/**
+	 * TODO: What is this? Are there specific values we expect?
+	 */
 	permission: string[];
+
+	/**
+	 * The user associated with this client.
+	 *
+	 * @remarks There may be more than 1 client associated with the same user.
+	 */
 	user: IUser;
+
+	/**
+	 * TODO: What is this? Are there specific values we expect?
+	 */
 	scopes: string[];
 
 	/**
-	 * The time the client connected
+	 * The time the client connected to the service.
+	 *
+	 * TODO: what does it mean for this to be undefined?
 	 */
 	timestamp?: number;
 }
 
+/**
+ * A client that has been acknowledged by the sequencer.
+ */
 export interface ISequencedClient {
+	/**
+	 * The underlying client details.
+	 */
 	client: IClient;
 
+	/**
+	 * The sequence number of the "join" message sent when the client joined the session.
+	 */
 	sequenceNumber: number;
 }
 
+/**
+ * TODO
+ */
 export interface ISignalClient {
+	/**
+	 * The {@link ISignalClient.client}'s unique ID.
+	 */
 	clientId: string;
 
+	/**
+	 * The underlying client details.
+	 */
 	client: IClient;
 
 	/**
-	 * Counts the number of signals sent by the client
+	 * Counts the number of signals sent by the client.
+	 *
+	 * TODO: What does it mean for this to be undefined?
 	 */
 	clientConnectionNumber?: number;
 
 	/**
-	 * Sequence number that indicates when the signal was created in relation to the delta stream
+	 * Sequence number that indicates when the signal was created in relation to the delta stream.
+	 *
+	 * TODO: What does it mean for this to be undefined?
```

Comment:
General feedback: In many cases where you post this question, there are two ways we can answer it (i.e. there is very good chance that nobody has an answer without doing one of the following):
- Tracing code.
- Posting sample values, you see in payloads (for example, the values for all properties in IClient), as that reminds people what's actually there.

Reply:
That's fair, but I also want to avoid documenting things in terms of how they're currently used. So my general hope is that folks with knowledge in the space can fill in the "what is this supposed to be" before resorting to using current usage patterns to inform the docs.

I also think the question "what does it mean for this to be undefined, null, etc." is an important question for devs to have in mind when adding optional properties / parameters. I would love to see us start documenting those meanings more thoroughly.

Reply:
I'm sure there are some number of people (including me) who can help here, but I simply do not remember a bunch of stuff :) Almost always I trace code to remind me what it is :)